src/bentoml/deployment.py

```diff
@@ -0,0 +1,444 @@
+        config_dct=config_dct,
+        config_file=config_file,
+        path_context=path_context,
+        context=context,
+        _bento_store=_bento_store,
+        _cloud_client=_cloud_client,
+    )
+
+
+@t.overload
+def apply(
+    name: str,
+    cluster_name: t.Optional[str] = ...,
+    path_context: t.Optional[str] = ...,
+    context: t.Optional[str] = ...,
+    _bento_store: BentoStore = Provide[BentoMLContainer.bento_store],
+    _cloud_client: BentoCloudClient = Provide[BentoMLContainer.bentocloud_client],
+    *,
+    project_path: t.Optional[str] = ...,
+    config_dct: t.Optional[dict[str, t.Any]] = ...,
+) -> Deployment:
+    ...
+
+
+@t.overload
+def apply(
+    name: str,
+    cluster_name: t.Optional[str] = ...,
+    path_context: t.Optional[str] = ...,
+    context: t.Optional[str] = ...,
+    _bento_store: BentoStore = Provide[BentoMLContainer.bento_store],
+    _cloud_client: BentoCloudClient = Provide[BentoMLContainer.bentocloud_client],
+    *,
+    bento: t.Optional[t.Union[Tag, str]] = ...,
+    config_dct: t.Optional[dict[str, t.Any]] = ...,
+) -> Deployment:
+    ...
+
+
+@t.overload
+def apply(
+    name: str,
+    cluster_name: t.Optional[str] = ...,
+    path_context: t.Optional[str] = ...,
+    context: t.Optional[str] = ...,
+    _bento_store: BentoStore = Provide[BentoMLContainer.bento_store],
+    _cloud_client: BentoCloudClient = Provide[BentoMLContainer.bentocloud_client],
+    *,
+    project_path: t.Optional[str] = ...,
+    config_file: t.Optional[str] = ...,
+) -> Deployment:
+    ...
+
+
+@t.overload
+def apply(
+    name: str,
+    cluster_name: t.Optional[str] = ...,
+    path_context: t.Optional[str] = ...,
+    context: t.Optional[str] = ...,
+    _bento_store: BentoStore = Provide[BentoMLContainer.bento_store],
+    _cloud_client: BentoCloudClient = Provide[BentoMLContainer.bentocloud_client],
+    *,
+    bento: t.Optional[t.Union[Tag, str]] = ...,
+    config_file: t.Optional[str] = ...,
+) -> Deployment:
+    ...
+
+
+@inject
+def apply(
+    name: str,
+    cluster_name: str | None = None,
+    path_context: str | None = None,
+    context: str | None = None,
+    _bento_store: BentoStore = Provide[BentoMLContainer.bento_store],
+    _cloud_client: BentoCloudClient = Provide[BentoMLContainer.bentocloud_client],
+    *,
+    project_path: str | None = None,
+    bento: Tag | str | None = None,
+    config_dct: dict[str, t.Any] | None = None,
+    config_file: str | None = None,
+) -> Deployment:
+    return Deployment.apply(
+        name=name,
+        project_path=project_path,
+        bento=bento,
+        cluster_name=cluster_name,
+        context=context,
+        path_context=path_context,
+        _bento_store=_bento_store,
+        _cloud_client=_cloud_client,
+        config_dct=config_dct,
+        config_file=config_file,
+    )
+
+
+def get(
+    name: str,
+    context: str | None = None,
+    cluster_name: str | None = None,
+    kube_namespace: str | None = None,
```

Comment:
I think we can delete `kube_namespace` parameter and only use default namespace to call the API.